src/main/scala/io/iohk/ethereum/network/p2p/MessageDecoders.scala

```diff
@@ -38,45 +35,87 @@ object NetworkMessageDecoder extends MessageDecoder {
 // scalastyle:off
 object EthereumMessageDecoder extends MessageDecoder {
 
-  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message =
+  private type MessageHandler = PartialFunction[(Int, Array[Byte]), Message]
```

Comment:
How is this approach better/cleaner than what I proposed?

Reply:
It is far more composable - with your approach I can't just do `case _ => handlePV63(msg, payload)` because it isn't true, we are handling NewBlock/Status differently than PV63, so it ends up with copying all those PV63 and PV62 messages or doing similar partition as I did and instead of `orElse` use `case _ => nextHandler(msg, payload)` where nextHandler will have another call with wildcard etc. For me it's just the perfect fit for PartialFunction

Reply:
> I can't just do case _ => handlePV63(msg, payload) because it isn't true

What isn't true?

Reply:
ok you right, I can do this with simple matching, I forgot that those new messages would be matched firstly and the bad path is not reachable